<a href="https://colab.research.google.com/github/Raizel820/Inferential_Statistics/blob/master/Project_Examine_Racial_Discrimination_Using_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
I will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and are submitted to my Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

It can be included written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution

In [0]:
%matplotlib inline
import pandas as pd

import subprocess as sp
import sys,os,shutil,pickle,time
import numpy as np
from os.path import join
from matplotlib import pyplot as plt
plt.style.use('default')
# mount Google Drive
from os.path import expanduser
gd_path=join(expanduser("~"),'gd')
if not os.path.isdir(gd_path):
    try:
        # load Google Drive
        from google.colab import drive,files
        drive.mount('/drive')
        sp.call('ln -s /drive/My\ Drive '+gd_path, shell=True)
    except:
        print('unable to find Google Drive Folder')
os.chdir(gd_path+'/MachineLearning')
os.listdir('.')

In [0]:
import pandas as pd
import numpy as np
from scipy import stats

In [0]:
df = pd.io.stata.read_stata('us_job_market_discrimination.dta')
df=data

In [0]:
# number of callbacks for black-sounding names
sum(data[df.race=='w'].call)

235.0

In [0]:
df.head()

id ad  education  ofjobs  ...  busservice  othservice  missind  ownership
0  b  1          4       2  ...         0.0         0.0      0.0           
1  b  1          3       3  ...         0.0         0.0      0.0           
2  b  1          4       1  ...         0.0         0.0      0.0           
3  b  1          3       4  ...         0.0         0.0      0.0           
4  b  1          3       3  ...         0.0         1.0      0.0  Nonprofit

[5 rows x 65 columns]

In [0]:
len(data)

4870

#### Q1) What test is appropriate for this problem? Does CLT apply?
CLT applies because we assume that the samples are representative of the population. The sample size is 4870. Literature typically suggests a lower limit of 30 observations in a sample for CLT to hold. In terms of CLT, the sample is large enough.

#### Q2) What are the null and alternate hypotheses?
Null hypotesis: Race does not have a impact on the rate of callbacks for resumes.

Alternative Hypotesis: Race has a significant impact on the rate of callbacks for resumes.

We have two categorical variables here: race and call. Chi-square is the appriate method for investitating the relation.


#### Q3) Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.


In [0]:
w = data[df.race=='w']
b = data[df.race=='b']

n_w = len(w)
n_b = len(b)

prop_w = np.sum(w.call) / len(w)
prop_b = np.sum(b.call) / len(b)

prop_diff = prop_w - prop_b
phat = (np.sum(w.call) + np.sum(b.call)) / (len(w) + len(b))

z = prop_diff / np.sqrt(phat * (1 - phat) * ((1 / n_w) + (1 / n_b)))
pval = stats.norm.cdf(-z) * 2
print("Z score: {}".format(z))
print("P-value: {}".format(pval))

Z score: inf
P-value: 0.0


/usr/local/lib/python2.7/dist-packages/ipykernel_launcher.py:13: RuntimeWarning: divide by zero encountered in double_scalars
  del sys.path[0]


In [0]:
prop_w


0.09650924024640657

In [0]:
prop_b 

0.06447638603696099

In [0]:
#Frequentist Approach
permutation_replicates = np.empty(100000)

white = data[data.race == 'w'].call.values
black = data[data.race == 'b'].call.values
diff_of_means = np.mean(white) - np.mean(black)

for i in range(len(permutation_replicates)):
    permutation_samples = np.random.permutation(np.concatenate((white, black)))
    
    white_perm = permutation_samples[:len(white)]
    black_perm = permutation_samples[len(white):]
    
    permutation_replicates[i] = np.abs(np.mean(white_perm) - np.mean(black_perm))

p = np.sum(permutation_replicates > diff_of_means) / len(permutation_replicates)
print('p =', p)

('p =', 0)


In [0]:
moe = 1.96 * np.sqrt(phat * (1 - phat) * ((1 / n_w) + (1 / n_b)))
ci = prop_diff + np.array([-1, 1]) * moe
print("Margin of Error: {}".format(moe))
print("Confidence interval: {}".format(ci))

Margin of Error: 0.0
Confidence interval: [0.03203285 0.03203285]


####  Q4) Write a story describing the statistical significance in the context or the original problem.

The results shows that there is significance relation between rate of callbacks and race. White people have higher call back rates compared to black people when they apply a job.
  
####  Q5) Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

No, my analysis does not imply that race is the most significant factor in callback success. We need to keep in mind that there can be other elements such as socio-cultural factors, socio-economic status etc. Taking a look at litetature on this field may provide us a better understanding about the nature of the issue. With a better knowledge on the subject, we can make more robust inferences. 